pkg/storage/postgres/schema.go

```diff
@@ -56,28 +56,46 @@ CREATE TABLE task_run (
 	id serial PRIMARY KEY,
 	vcs_id integer REFERENCES vcs(id),
 	task_id integer NOT NULL REFERENCES task (id) ON DELETE CASCADE,
+	total_input_digest text NOT NULL,
 	start_timestamp timestamp with time zone NOT NULL,
 	stop_timestamp timestamp with time zone NOT NULL,
 	result text NOT NULL,
 	CONSTRAINT result_check CHECK (result in ('success', 'failure'))
 );
+CREATE INDEX idx_task_run_total_input_digest ON task_run(total_input_digest);
 
-CREATE TABLE input (
+CREATE TABLE input_file (
 	id serial PRIMARY KEY,
-	uri text NOT NULL,
+	path text NOT NULL,
+	digest text NOT NULL,
+	CONSTRAINT input_file_path_digest_uniq UNIQUE (path, digest)
+);
+CREATE INDEX idx_input_file_path ON input_file(path);
+
+CREATE TABLE input_string (
+	id serial PRIMARY KEY,
```

Comment:
optional: since we use `digest` on an unique constraint, we could use the digest as the primary key and get rid of the serial column.

Reply:
Yes, I'm using a numerical PK because it only uses 4 Bytes whenever it is stored as FK in other tables. 
I also expect that queries are faster when the FK is only 4B, I don't know that for sure though :-)

Reply:
Good points! I was not considering those.

> For searches, the current approach requires to always join the the table on a query, even if you only want the digest

Right, currently we do not have this usecase in baur.

> Only searches that read the text of the string parameter will be improved by the smaller size of the column used on the relationship.

I would expect also Joins because the key is much smaller.
I have no clue what differences that does in practice, probably not noticeable until  there are giant amounts of records in the tables  :-)

I will keep it as it is for now.